coral-incremental/src/main/java/com/linkedin/coral/incremental/RelNodeCostEstimator.java

```diff
@@ -0,0 +1,237 @@
+      this.leftTableName = leftTableName;
+      this.rightTableName = rightTableName;
+      this.leftFieldName = leftFieldName;
+      this.rightFieldName = rightFieldName;
+    }
+  }
+
+  private Map<String, Double> stat = new HashMap<>();
+
+  private Map<String, Double> distinctStat = new HashMap<>();
+
+  public void setStat(Map<String, Double> stat) {
+    this.stat = stat;
+  }
+
+  public void setDistinctStat(Map<String, Double> distinctStat) {
+    this.distinctStat = distinctStat;
+  }
+
+  private Double IOCostParam = 1.0;
+
+  private Double shuffleCostParam = 1.0;
+
+  public void setIOCostParam(Double IOCostParam) {
+    this.IOCostParam = IOCostParam;
+  }
+
+  public void setShuffleCostParam(Double shuffleCostParam) {
+    this.shuffleCostParam = shuffleCostParam;
+  }
+
+  public void loadStatistic(String configPath) {
+    try {
+      String content = new String(Files.readAllBytes(Paths.get(configPath)));
+      // Parse JSON string to JsonObject
+      JsonObject jsonObject = new JsonParser().parse(content).getAsJsonObject();
+      // Iterate over each table in the JSON object
+      for (Map.Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+        String tableName = entry.getKey();
+        JsonObject tableObject = entry.getValue().getAsJsonObject();
+
+        // Extract row count
+        Double rowCount = tableObject.get("RowCount").getAsDouble();
+
+        // Extract distinct counts
+        JsonObject distinctCounts = tableObject.getAsJsonObject("DistinctCounts");
+
+        stat.put(tableName, rowCount);
+
+        // Iterate over distinct counts
+        for (Map.Entry<String, JsonElement> distinctEntry : distinctCounts.entrySet()) {
+          String columnName = distinctEntry.getKey();
+          Double distinctCount = distinctEntry.getValue().getAsDouble();
+
+          distinctStat.put(tableName + ":" + columnName, distinctCount);
+        }
+
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+  }
+
+  public Double getCost(RelNode rel) {
+    CostInfo executionCostInfo = getExecutionCost(rel);
+    Double IOCost = executionCostInfo.row * IOCostParam;
+    return executionCostInfo.cost * shuffleCostParam + IOCost;
+  }
+
+  public CostInfo getExecutionCost(RelNode rel) {
+    if (rel instanceof TableScan) {
+      return getExecutionCostTableScan((TableScan) rel);
+    } else if (rel instanceof LogicalJoin) {
+      return getExecutionCostJoin((LogicalJoin) rel);
+    } else if (rel instanceof LogicalUnion) {
+      return getExecutionCostUnion((LogicalUnion) rel);
+    } else if (rel instanceof LogicalProject) {
+      return getExecutionCostProject((LogicalProject) rel);
+    }
+    return new CostInfo(0.0, 0.0);
+  }
+
+  private CostInfo getExecutionCostTableScan(TableScan scan) {
+    RelOptTable originalTable = scan.getTable();
+    String tableName = getTableName(originalTable);
+    Double row = stat.getOrDefault(tableName, 5.0);
```

Comment:
Would it be smarter to skip over the table if we have no statistics about it? Using a default value could throw off the final cost by a significant magnitude.

If we're keeping, please make `5.0` a default constant variable.